moto/appmesh/utils/spec_parsing.py

```diff
@@ -0,0 +1,681 @@
+from typing import Any, Dict, List, Optional
+
+from moto.appmesh.dataclasses.route import (
+    GrcpRouteRetryPolicy,
+    GrpcMetadatum,
+    GrpcRoute,
+    GrpcRouteMatch,
+    HttpRoute,
+    HttpRouteMatch,
+    HttpRouteRetryPolicy,
+    Match,
+    QueryParameterMatch,
+    Range,
+    RouteAction,
+    RouteActionWeightedTarget,
+    RouteMatchPath,
+    RouteMatchQueryParameter,
+    RouteSpec,
+    TCPRoute,
+    TCPRouteMatch,
+)
+from moto.appmesh.dataclasses.shared import Duration, Timeout
+from moto.appmesh.dataclasses.virtual_node import (
+    ACM,
+    DNS,
+    SDS,
+    AccessLog,
+    AccessLogFile,
+    AWSCloudMap,
+    Backend,
+    BackendDefaults,
+    BackendTrust,
+    Certificate,
+    CertificateFile,
+    CertificateFileWithPrivateKey,
+    ClientPolicy,
+    ConnectionPool,
+    GRPCOrHTTP2Connection,
+    HealthCheck,
+    HTTPConnection,
+    KeyValue,
+    Listener,
+    ListenerCertificateACM,
+    ListenerTLS,
+    Logging,
+    LoggingFormat,
+    OutlierDetection,
+    PortMapping,
+    ProtocolTimeouts,
+    ServiceDiscovery,
+    SubjectAlternativeNames,
+    TCPConnection,
+    TCPTimeout,
+    TLSBackendValidation,
+    TLSClientPolicy,
+    TLSListenerCertificate,
+    TLSListenerValidation,
+    Trust,
+    VirtualNodeSpec,
+    VirtualService,
+)
+from moto.appmesh.dataclasses.virtual_node import (
+    Match as VirtualNodeMatch,
+)
+from moto.appmesh.dataclasses.virtual_router import PortMapping as RouterPortMapping
+from moto.appmesh.exceptions import (
+    MissingRequiredFieldError,
+)
+
+
+def clean_dict(obj: Dict[str, Any]) -> Dict[str, Any]:  # type: ignore[misc]
+    return {
```

Comment:
This method isnt' used anywhere - looks like the duplicate method in `utils.py` is used instead.